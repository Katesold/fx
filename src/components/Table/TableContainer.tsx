import { useContext, useState } from "react";
import { Button } from "../Button/Button"
import { Table } from "./Table"
import { TableProps } from "./types"
import { StyledTableContainer } from './styles';
import { AppContext } from "../../App";


export const TableContainer = () => {
    const ctxData = useContext(AppContext);
    const [isValid, setIsValid] = useState(false);
    return (
        <StyledTableContainer>
            <Table setIsValid={setIsValid} />
            {isValid && <Button text='Submit' />}
        </StyledTableContainer>
    )
}