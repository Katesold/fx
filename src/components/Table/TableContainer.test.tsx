import { TableContainer } from './TableContainer';
import { screen, render, waitFor } from '@testing-library/react';
import { toast } from 'react-toastify';
import userEvent from '@testing-library/user-event';

jest.mock('react-toastify');

describe('TableContainer components are being tested', () => {
    beforeEach(() => jest.clearAllMocks());
    const props = {
        data: ['test', 'test2', '23555', '4w55']
    };
    it('should match the snapshot', () => {
        const { container } = render(<TableContainer {...props}/>);
        expect(container).toMatchSnapshot();
        expect(screen.getByRole('button')).toBeInTheDocument();
    });

    it('should trigger toast success', async () => {
        render(<TableContainer {...props}/>);
        const button = screen.getByRole('button');
        expect(button).toBeInTheDocument();
        userEvent.click(button);
        await waitFor(() => {
            expect(toast.success).toBeCalledWith(expect.anything(), {"className": "toastSuccessBg", "hideProgressBar": true});
            expect(toast.success).toBeCalledWith('Payments Uploaded', {"className": "toastSuccessBg", "hideProgressBar": true});
            expect(toast.success).toBeCalledTimes(2);
          });
    });
    it('should match the snapshot with incorrect data', () => {
        const wrongData = {
            data: [
                "Beneficiary ID,Payment Currency,Payment Amount,Payment Reference",
                ",GBP,122,Invoice 4",
                "100005,GBP,255,Invoice 5",
                ""
            ]
        }
        const { container } = render(<TableContainer {...wrongData}/>);
        expect(container).toMatchSnapshot();
        expect(screen.queryByRole('button')).not.toBeInTheDocument();
    });
});