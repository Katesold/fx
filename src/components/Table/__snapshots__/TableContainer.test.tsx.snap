// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableContainer components are being tested should match the snapshot 1`] = `
<div>
  <div
    class="sc-gswNZR joPihG"
  >
    <table
      class="sc-dkrFOg dgDqgz"
    >
      <thead
        class="sc-hLBbgP bzkRai"
      >
        <tr
          class="sc-gKPRtg fXGJDB"
        >
          <th
            class="sc-jSUZER foWzlI"
          >
            test
          </th>
        </tr>
      </thead>
      <tbody
        class="sc-eDvSVe bJxIbO"
      >
        <tr
          class="sc-iBYQkv gaauiS"
        >
          <td
            class="sc-ftTHYK jdUNiY"
          >
            test2
          </td>
        </tr>
        <tr
          class="sc-iBYQkv gaauiS"
        >
          <td
            class="sc-ftTHYK jdUNiY"
          >
            23555
          </td>
        </tr>
        <tr
          class="sc-iBYQkv gaauiS"
        >
          <td
            class="sc-ftTHYK jdUNiY"
          >
            4w55
          </td>
        </tr>
      </tbody>
    </table>
    <button
      class="sc-bcXHqe eXfJSU"
      color="green"
    >
      Submit
    </button>
  </div>
</div>
`;

exports[`TableContainer components are being tested should match the snapshot with incorrect data 1`] = `
<div>
  <div
    class="sc-gswNZR joPihG"
  >
    <table
      class="sc-dkrFOg dgDqgz"
    >
      <thead
        class="sc-hLBbgP bzkRai"
      >
        <tr
          class="sc-gKPRtg fXGJDB"
        >
          <th
            class="sc-jSUZER foWzlI"
          >
            Beneficiary ID
          </th>
          <th
            class="sc-jSUZER foWzlI"
          >
            Payment Currency
          </th>
          <th
            class="sc-jSUZER foWzlI"
          >
            Payment Amount
          </th>
          <th
            class="sc-jSUZER foWzlI"
          >
            Payment Reference
          </th>
        </tr>
      </thead>
      <tbody
        class="sc-eDvSVe bJxIbO"
      >
        <tr
          class="sc-iBYQkv gaauiS"
        >
          <td
            class="sc-ftTHYK jdUNiY"
          >
            100005
          </td>
          <td
            class="sc-ftTHYK jdUNiY"
          >
            <img
              alt="currency"
              src="flag.png"
            />
            GBP
          </td>
          <td
            class="sc-ftTHYK jdUNiY"
          >
            255
          </td>
          <td
            class="sc-ftTHYK jdUNiY"
          >
            Invoice 5
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
